packages/experimental/theme/README.md

```diff
@@ -0,0 +1,60 @@
+
+The issue is that this layering adds overhead as well. When building HOCs the number of elements in the JSX tree, can be far greater than the actual number of primitives rendered in the DOM. While it is possible to render a function component by calling it directly, this has to be done with caution if that component contains hooks. The storage for hooks is effectively an array indexed from the last `createElement` call. If a `useState` call were to happen on one render pass, but be skipped in the next, all subsequent hook calls would be misaligned.
+
+## Staged Components
+
+To solve this reliably for function components it is necessary to separate the hook calls from element tree generation. The pattern in this package is to write render functions that return a continuation function that will return the JSX tree. This pattern is as follows:
+
+```tsx
+const twoPartRender = (props: MyProps) => {
+  // do the hook calls in this section */
+  const theme = React.useContext(ThemeContext);
+  const [state, setState] = React.useState(() => doSomeStateSetup()));
+
+  // now return a (props) or (props, children) or (props, ...children) function that finishes the render
+  return (additional: MyProps, ...children: React.ReactNode[]) => {
+    const merged = { ...props, ...additional };
+    return (<View {...merged}>{children}</View>)
+  }
+}
+```
+
+### `stagedComponent`
+
+This type of function is not recognizable on its own as a component. This package exports a helper function `stagedComponent` that both:
+
+- Turns a two part render function into a component that react can recognize
+- Adds the means for the framework to access the initial two part function if it knows how to handle it
+
+The helper will return a `React.FunctionComponent` that will forward props (without children) to the first call, then pass children to the continuation function.
+
+## Consuming staged components
+
+Staged components could be consumed by hand. This might look something like:
+
+```ts
+// create the BaseComponent as a staged component, this can still be rendered via <BaseComponent />
+const BaseComponent = stagedComponent(/* some implementation */);
+
+// create a wrapper component
+const WrapperComponent = stagedComponent((props) => {
+  // grab the children, do some prop transformations, etc.
+  const { children, ...rest } = props;
+  const propsForBase = doSomethingWithProps(rest);
+  // call the first part of the function
+  const continuation = BaseComponent._staged(propsForBase);
+  // now render that component just by calling the function
+  return continuation({}, children)
+}
+```
+
+Note that in this example the `WrapperComponent` is leveraging the code of the `BaseComponent` but will not create a dedicated tree entry. Also the `WrapperComponent` itself could be implemented as a staged component.
```

Comment:
If I'm understanding this correctly, what `stagedComponent` give us is a way to compress from n (where n > 2) layered components in the render tree to 2. So if `BaseComponent` renders a `View` in it's continuation `WrapperComponent` looks like
```
<WrapperComponent>
      <View>
            ....
      </View>
</WrapperComponent>
```

So is this pulling out some of the ideas from the original `compose` and `composable` concepts into their own helpers?

Reply:
Yes, that's the way the experimental framework is built. The pieces can be used on their own, or together. Compose brings them all together, there is no more vanilla composable (which was confusing to folks).

That being said, this is the existing documentation file in use-slots, which accidentally got copied when I created the new package.

Reply:
Makes sense :)